src/main/java/hexgui/util/ProcessUtil.java

```diff
@@ -0,0 +1,106 @@
+// ProcessUtil.java
+
+package hexgui.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+class ExitWaiter extends Thread {
+  public ExitWaiter(Object monitor, Process process) {
+    m_monitor = monitor;
+    m_process = process;
+  }
+
+  public boolean isFinished() {
+    synchronized (m_mutex) {
+      return m_isFinished;
+    }
+  }
+
+  public void run() {
+    try {
+      m_process.waitFor();
+    } catch (InterruptedException e) {
+    }
+    synchronized (m_mutex) {
+      m_isFinished = true;
+    }
+    synchronized (m_monitor) {
+      m_monitor.notifyAll();
+    }
+  }
+
+  private boolean m_isFinished;
+
+  private final Object m_monitor;
+
+  private final Object m_mutex = new Object();
+
+  private final Process m_process;
+}
+;
+
+/** Static utility functions and classes related to processes. */
+public class ProcessUtil {
+  /** Copies standard error of a process to System.err. */
+  public static class StdErrThread extends Thread {
+    public StdErrThread(Process process) {
+      super(new StreamCopy(false, process.getErrorStream(), System.err, false));
+    }
+  }
+
+  /** Run a process and return its standard output as a string. */
+  public static String runCommand(String[] cmdArray) throws IOException {
+    Runtime runtime = Runtime.getRuntime();
+    Process process = runtime.exec(cmdArray);
+    Thread discardErr = new StreamDiscard(process.getErrorStream());
+    discardErr.start();
+    InputStream in = process.getInputStream();
+    BufferedReader reader = new BufferedReader(new InputStreamReader(in));
+    try {
+      StringBuilder result = new StringBuilder();
+      String line;
+      while ((line = reader.readLine()) != null) {
+        result.append(line);
+        result.append('\n');
+      }
+      try {
+        if (process.waitFor() != 0) throw new IOException("Process returned error status");
+      } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)